### PR TITLE
show on shared boards unassigned cards to all users

### PR DESCRIPTION
**Pull request by jakobroehrl**
Signed-off-by: Jakob Röhrl <jakob.roehrl@web.de>

* Resolves: #3076
* Target version: master

### CODE REVIEW

File: lib/Db/CardMapper.php [hunk 1]
```diff
@@ -236,10 +236,12 @@ public function findAssignedCards($boardId, $username) {
...
 			->innerJoin('s', 'deck_boards', 'b', 'b.id = s.board_id')
-			->innerJoin('c', 'deck_assigned_users', 'u', 'c.id = u.card_id')
+			->leftJoin('c', 'deck_assigned_users', 'u', 'c.id = u.card_id')
 			->where($qb->expr()->eq('s.board_id', $qb->createNamedParameter($boardId, IQueryBuilder::PARAM_INT)))
-			->andWhere($qb->expr()->eq('u.participant', $qb->createNamedParameter($username, IQueryBuilder::PARAM_STR)))
-			->andWhere($qb->expr()->eq('u.type', $qb->createNamedParameter(Acl::PERMISSION_TYPE_USER, IQueryBuilder::PARAM_INT)))
+			->andWhere($qb->expr()->orX(
+				$qb->expr()->eq('u.participant', $qb->createNamedParameter($username, IQueryBuilder::PARAM_STR)),
+				$qb->expr()->isNull('u.participant'))
+			)
```
Comment (juliushaertl) [inline lib/Db/CardMapper.php:244]: As mentioned in the chat, the method should be renamed to also indicate that unassigned cards are fetched as well.

File: lib/Service/CardService.php [hunk 2]
```diff
@@ -615,8 +615,8 @@ public function findAllWithDue($userId) {
 	 * @throws \OCA\Deck\NoPermissionException
 	 * @throws BadRequestException
 	 */
-	public function findAssignedCards($userId) {
-		$cards = $this->cardMapper->findAssignedCards($userId);
+	public function findToMeOrNotAssignedCards($userId) {
+		$cards = $this->cardMapper->findToMeOrNotAssignedCards($userId);
```
Comment (juliushaertl) [inline lib/Service/CardService.php:619]: We actually don't use that method anywhere so we can just drop it for now, it didn't match the previous method signature as well.

Comment (juliushaertl) [reply lib/Service/CardService.php:619]: That will also fix the psalm error then

**Review by juliushaertl — APPROVED** [review]

**Comment by stefan-niedermann** [comment]
@juliushaertl can you tell me the exact Deck versions where this will be published? If i remember correctly, you are maintaining multiple versions in parallel, right?

**Comment by juliushaertl** [comment]
Yes, currently this is only scheduled for the next release (see milestone). Basically all PRs merged to master will go to the next major. For maintenance releases PRs would need to be backported to the stable1.x branches.

**Comment by juliushaertl** [comment]
@stefan-niedermann I changed plans here a bit since I'll push out a Nextcloud 22 compatible release ans as there are no major changes included we can keep the app compatbile to both 21 and 22. So the next version will be 1.5 including that change while any below that will not have it.

**Comment by stefan-niedermann** [comment]
Thanks for the update!

**Pull request merged**